pkgs/applications/science/astronomy/gildas/format-security.patch

```diff
@@ -0,0 +1,32 @@
+diff -u -r gildas-src-feb17d.orig/kernel/lib/gcore/gcomm.c gildas-src-feb17d/kernel/lib/gcore/gcomm.c
+--- gildas-src-feb17d.orig/kernel/lib/gcore/gcomm.c	2017-05-18 00:23:07.482284541 +0200
++++ gildas-src-feb17d/kernel/lib/gcore/gcomm.c	2017-05-18 00:24:03.043396641 +0200
+@@ -973,12 +973,12 @@
+     i = 0;
+     sprintf( exec_argv[i++], "xterm");
+     sprintf( exec_argv[i++], "-T");
+-    sprintf( exec_argv[i++], app_name);
++    sprintf( exec_argv[i++], "%s", app_name);
+     sprintf( exec_argv[i++], "-e");
+     sprintf( exec_argv[i++], "gdb");
+     sprintf( exec_argv[i++], "-x");
+-    sprintf( exec_argv[i++], gdb_cmd_file);
+-    sprintf( exec_argv[i++], app_name);
++    sprintf( exec_argv[i++], "%s", gdb_cmd_file);
++    sprintf( exec_argv[i++], "%s", app_name);
+     exec_argv[i++] = NULL;
+     if (i >= MAX_ARGC) {
+         gcore_c_message(seve_e, "SIC", "Too much arguments, exiting");
```

Comment:
much -> many

Also please output how many arguments were given and how many was the maximum expected.